### ISSUE TITLE
Building issues on Visual Studio 2019

### ISSUE THREAD

**Issue by hedin-hiervard** [aantron/better-enums#75]
Just tried building the old project (which built perfectly in 2015 and 2017) and build with the latest XCode as well.
I got a bunch of errors like these:
```
>Z:\Projects\_ARCHIVE\PK3\src\util/EnumArray.h(19): error C2975: '_Size': invalid template argument for 'std::array', expected compile-time constant expression
1>C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\array(20): note: see declaration of '_Size'

```

Seems like `::_size()` method is not `constexpr` (BETTER_ENUMS_HAVE_CONSTEXPR is defined).

all of them seem to be related to better-enums. I tried toolsets 140 (which was fine with 2017) and 142 and C++ standards 14, 17 and latest with little luck.

**Comment by aantron** [comment]
How is `_Size` is defined? I don't believe this is from `enum.h`.

Is this with the latest `enum.h`?

**Comment by aantron** [comment]
Closing, please reopen if there is more info.

**Comment by helgeklein** [comment]
I got a similar error from the compiler when I tried to add a `BETTER_ENUM` as non-type template parameter. I solved it by instantiating global variables with external linkage and use them as template params instead. See the relevant code below:

Enum definition (in .h file):

    BETTER_ENUM (DataType, int, Date = 1, String, Int, Double, Bool)

Variable declaration (in .h file):

    extern const DataType dataTypeString;

Variable definition (in .cpp file):

    const DataType dataTypeString = DataType::String;

Class template (in .h file):

    template <const DataType& FieldType, typename T>
    class Field
    {
       ...
    {

Utilizing the template above in a derived class (in .h file):

    class StringField : public Field<dataTypeString, std::string>
    {
       ...
    {

I hope this helps someone else (or my future self).
